src/WebJobs.Extensions.DurableTask/AzureStorageDurabilityProvider.cs

```diff
@@ -54,6 +52,10 @@ internal class AzureStorageDurabilityProvider : DurabilityProvider
 
         public override JObject ConfigurationJson => this.storageOptionsJson;
 
+        public override TimeSpan MaximumDelayTime { get; set; } = TimeSpan.FromDays(6);
+
+        public override TimeSpan LongRunningTimerIntervalLength { get; set; } = TimeSpan.FromDays(3);
```

Comment:
Setting this to 3 days means that in-flight orchestrations that have timers of lengths between 3 and 6 days (the current maximum) will end up with non-determinism errors. 

In order to mitigate this, we have a few options:

1. Make long-running timers opt-in. This means adding a flag on DurableTaskOptions, and will probably greatly reduce the number of customers who make use of the feature.
2. Make a note as a "Breaking Change" in the release notes. We generally try to avoid breaking changes for in-flight orchestrations, as many customers don't read the release notes, and might shoot themselves in the foot this way.
3. Set the timer interval to 6 days, as no customers have timers currently longer than 6 days due to previous limits in place.

@cgillum thoughts?

Reply:
@ConnorMcMahon I don't yet understand why this change would cause non-determinism failures for in-flight orchestrations. Are you saying that this PR will cause us to change the fire time for already scheduled timers? I didn't get that impression when reading the code. My understanding was that this 3 day interval value only applies to timers that are scheduled for longer than the 6 day maximum. Otherwise we don't behave differently. If I'm misunderstanding the logic and you are certain about this problem, then I'm totally fine with option (3).

Reply:
Ah I misread this code in this approach. I think you are right, as the internalFireAt should equal fireAt if we don't exceed the maximum timer. That makes sense, and completely nullifies this concern.